text/0000-reserve-element.md

```diff
@@ -0,0 +1,55 @@
+- Start Date: 2015-12-03
+- RFC PR: (leave this empty)
+- Refract Issue: (leave this empty)
+
+# Summary
+
+This proposes we reserve the keyword `element` for all objects in a Refract
+tree.
+
+# Motivation
+
+This allows us to not worry about domain-specific knowledge when parsing a
+Refract tree. Right now, we have to rely on something out of band to tell us
+when something MAY be refracted, but if we rely on the reserved keyword
+`element`, we can know if an object is a normal object or a Refract Object.
+
+Note: we are already doing this in Minim.
+
+# Design
+
+This RFC proposes to reserve `element` in the entire tree. This means that
+anywhere the `element` keyword is found, a parser SHOULD treat the object as a
+Refract Object.
+
+## Secondary Proposal
+
+This allows for us to know when an object is a Refract Object or not. It also
+allows for us only refract things that need to be refracted when serializing.
+In this situation, for example, the `key` property of a `Member Element` would
+not need to refracted if it is a string.
```

Comment:
> ...things that need to be refracted when serializing.

We may want to define here exactly what that means. For example, we *must* refract `member` elements or expect implementations to handle them some other way (easy in `object`'s content, hard in attributes like `hrefVariables` for example). Maybe a whitelist of types supported to not be refracted? Does that make sense?